File: LeetCode_ProblemSet/Problem2033.cpp
#include <vector>
#include <algorithm>

class Solution {
public:
    int minOperations(std::vector<std::vector<int>>& grid, int x) {
        std::vector<int> nums;
        nums.reserve(grid.size() * grid[0].size());
        for (const auto& row : grid) {
            for (const auto num : row) {
                nums.push_back(num);
            }
        }
        std::sort(nums.begin(), nums.end());

        int cost = 0, median = nums[nums.size() / 2];
        for (const auto num : nums) {
            if ((median - num) % x != 0) return -1;
            cost += std::abs(median - num) / x; 
        }
        return cost;
    }
};

File: LeetCode_ProblemSet/Rust/src/Problem2033.rs
impl Solution {
    pub fn min_operations(grid: Vec<Vec<i32>>, x: i32) -> i32 {
        let mut nums: Vec<i32> = vec![];
        for row in grid {
            for num in row {
                nums.push(num);
            }
        }

        nums.sort();
        let (mut cost, median) = (0, nums[nums.len() / 2]);
        for num in nums {
            if (median - num) % x != 0 {
                return -1;
            }

            if median > num {
                cost += (median - num) / x;
            } else {
                cost += (num - median) / x;
            }
        }
        return cost;
    }
}